### PR TITLE
Fetch Retry Requests

### LINKED ISSUE #28: Retry fetch option for intermittent API errors
Excellent github action! I'm using your action to cache an external API which is secured with OAuth.

This API often experiences issues: `There was an error fetching from the API: FetchError: invalid json response body at https://example.com reason: Unexpected end of JSON input`. 

By retrying the same request in a few seconds the error will usually go away. I propose adding an option that allows failed requests to be retried N times waiting M seconds between requests.

### LINKED ISSUE #34: Cross Publish to GitHub Registry
This is planned, I've attempted to implement this in another repository of mine but I'm seeing an issue.

Related Issue: https://github.com/JamesIves/github-pages-deploy-action/issues/229

### LINKED ISSUE #35: Improve debugging

### PR DESCRIPTION
**Pull request by JamesIves**
**Description**
> Provide a description of what your changes do.

Adds the ability to pass in `RETRY: true` to the configuration, making any failed requests retry before failing. If set to true the requests will be attempted  one time, and then retried three more times. This will occur on the token and default endpoint requests until the job either fails or completes.

Also adds cross publishing support to the GitHub registry within `publish.yml`. 

**Testing Instructions**
> Give us step by step instructions on how to test your changes.

* Set `RETRY` to `true`.
* Attempt to run a failed API request: You can see some failed logs here (The logging has since been cleaned up): https://github.com/JamesIves/fetch-api-data-action/runs/552234486?check_suite_focus=true

**Additional Notes**
> Anything else that will help us test the pull request.

* I've also switched `console.log` to `info` as it's part of the actions toolkit suite.

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -1,28 +1,34 @@
...
-      - run: npm run-script build
-      - run: npm version patch -m "Release %s"
-      - run: npm ci
-      - run: npm publish
-        env:
-          NODE_AUTH_TOKEN: ${{ secrets.npm_token }}
\ No newline at end of file
+    - uses: actions/checkout@v2
+    - name: Configure git
+      run: |
```
Comment (lizafloyd) [inline .github/workflows/publish.yml:11]: What does this bar do here?

File: README.md [hunk 2]
```diff
@@ -180,6 +180,7 @@ The following configuration options should be set.
...
 | `TOKEN_CONFIGURATION` | Any applicable configuration settings that should be set such as authentication tokens. You can reference secrets using the `${{ secrets.secret_name }}` syntax. For more information refer to the [Fetch API documentation](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch).                                                                                               | `secrets / with` | **No**  |
+| `RETRY` | If you're working with an intermittent API you can toggle this option to `true`. Doing so will make the action try the request 3 times before failing.                                                                                                | `with` | **No**  |
```
Comment (lizafloyd) [inline README.md]: Is there any time between retries or will it try three times in a row? Might include that info in this

**Review by lizafloyd — COMMENTED** [review]
Sorry to be slow, where in your code does it increment the numbers of retries it's done?

File: .github/workflows/publish.yml (hunk 1 above)
Comment (JamesIves) [reply .github/workflows/publish.yml:11]: Executes a command on several lines vs a single one

File: README.md (hunk 2 above)
Comment (JamesIves) [reply README.md]: Good point! It's 3 times on a random interval.

**Pull request merged**